pkg/agent/controller/packetcapture/packetcapture_controller_linux.go

```diff
@@ -0,0 +1,879 @@
+	"k8s.io/client-go/util/workqueue"
+	klog "k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/agent/config"
+	"antrea.io/antrea/pkg/agent/controller/packetcapture/bpf"
+	"antrea.io/antrea/pkg/agent/interfacestore"
+	"antrea.io/antrea/pkg/agent/util"
+	crdv1alpha1 "antrea.io/antrea/pkg/apis/crd/v1alpha1"
+	clientsetversioned "antrea.io/antrea/pkg/client/clientset/versioned"
+	crdinformers "antrea.io/antrea/pkg/client/informers/externalversions/crd/v1alpha1"
+	crdlisters "antrea.io/antrea/pkg/client/listers/crd/v1alpha1"
+	binding "antrea.io/antrea/pkg/ovs/openflow"
+	"antrea.io/antrea/pkg/util/env"
+	"antrea.io/antrea/pkg/util/ftp"
+)
+
+type StorageProtocolType string
+
+const (
+	sftpProtocol StorageProtocolType = "sftp"
+)
+
+const (
+	controllerName               = "AntreaAgentPacketCaptureController"
+	resyncPeriod   time.Duration = 0
+
+	minRetryDelay = 5 * time.Second
+	maxRetryDelay = 300 * time.Second
+
+	defaultWorkers = 4
+
+	maxWaitingCaptures = 32
+
+	// reason for timeout
+	captureTimeoutReason   = "PacketCapture timeout"
+	defaultTimeoutDuration = time.Second * time.Duration(60)
+
+	captureStatusUpdatePeriod = 10 * time.Second
+
+	// PacketCapture uses a dedicated secret object to store auth info for file server.
+	// #nosec G101
+	fileServerAuthSecretName = "antrea-packetcapture-fileserver-auth"
+
+	// Max packet size for pcap capture.
+	snapshotLen = 65536
+)
+
+var (
+	packetDirectory = getPacketDirectory()
+	defaultFS       = afero.NewOsFs()
+)
+
+func getPacketDirectory() string {
+	return filepath.Join(os.TempDir(), "antrea", "packetcapture", "packets")
+}
+
+type packetCaptureState struct {
+	// name is the PacketCapture name.
+	name string
+	// numCapturedPackets record how many packets have been captured. Due to the RateLimiter,
+	// this maybe not be realtime data.
+	numCapturedPackets int32
+	// maxNumCapturedPackets is target number limit for our capture. When numCapturedPackets == maxNumCapturedPackets, it means that
+	// the PacketCapture is done successfully.
+	maxNumCapturedPackets int32
+	// updateRateLimiter controls the frequency of the updates to PacketCapture status.
+	updateRateLimiter *rate.Limiter
+	// pcapngFile is the file object for the packet file.
+	pcapngFile afero.File
+	// pcapngWriter is the writer for the packet file.
+	pcapngWriter *pcapgo.NgWriter
+}
+
+type Controller struct {
+	kubeClient            clientset.Interface
+	crdClient             clientsetversioned.Interface
+	packetCaptureInformer crdinformers.PacketCaptureInformer
+	packetCaptureLister   crdlisters.PacketCaptureLister
+	packetCaptureSynced   cache.InformerSynced
+	interfaceStore        interfacestore.InterfaceStore
+	nodeConfig            *config.NodeConfig
+	queue                 workqueue.TypedRateLimitingInterface[string]
+	sftpUploader          ftp.Uploader
+	newPacketSourceFn     func(c *Controller, pc *crdv1alpha1.PacketCapture) (PacketSource, error)
+	cond                  *sync.Cond
+	captures              map[string]string
```

Comment:
Can you add a comment for this `capture`, what does the key and value represent for?
Make sure to add a description to each field if it can help reviewers/developers to understand the struct and usage quicker. Please refer to other controllers' codes.

Reply:
ok